eng/targets/FunctionalTestWithAssets.targets

```diff
@@ -0,0 +1,27 @@
+<Project>
+
+  <ItemGroup>
+    <ProjectReference Include="@(FunctionalTestAssetProjectReference)" ReferenceOutputAssembly="false" />
```

Comment:
Why are the `@(FunctionalTestAssetProjectReference)` items built _at all_ on the build agents❔ When tests depend on newly-built AspNetCore.App bits, usual approach is to build them only on the Helix agents, adding `@(Content)` items for the test assets so they're part of the work item payload automatically. If necessary (see project template tests), fallbacks for local testing are of course fine. But, this seems to be set up for CI testing ☹️ 

And, why is this infrastructure centralized (it's only used in the gRPC `InteropTests` project)❔